drivers/wifi/wfx200/wfx200_host.c

```diff
@@ -0,0 +1,730 @@
+		break;
+	case WFM_STATUS_CONNECTION_TIMEOUT:
+		LOG_ERR("Connection timeout");
+		break;
+	case WFM_STATUS_CONNECTION_REJECTED_BY_AP:
+		LOG_ERR("Connection rejected by the access point");
+		break;
+	case WFM_STATUS_CONNECTION_AUTH_FAILURE:
+		LOG_ERR("Connection authentication failure");
+		break;
+	default:
+		LOG_ERR("Connection attempt error");
+	}
+	event->ev = WFX200_CONNECT_FAILED_EVENT;
+	k_queue_append(&wfx200_0->event_queue, event);
+}
+
+void sl_wfx_disconnect_callback(sl_wfx_disconnect_ind_t *disconnect)
+{
+	ARG_UNUSED(disconnect);
+	struct wfx200_queue_event *event = k_heap_alloc(&wfx200_0->heap,
+							sizeof(struct wfx200_queue_event), K_NO_WAIT);
+
+	if (event == NULL) {
+		/* TODO */
+		return;
+	}
+
+	wfx200_0->sl_context.state &= ~SL_WFX_STA_INTERFACE_CONNECTED;
+	event->ev = WFX200_DISCONNECT_EVENT;
+	k_queue_append(&wfx200_0->event_queue, event);
+
+	LOG_INF("WiFi disconnected");
+}
+
+void sl_wfx_start_ap_callback(sl_wfx_start_ap_ind_t *start_ap)
+{
+	struct wfx200_queue_event *event = k_heap_alloc(&wfx200_0->heap,
+							sizeof(struct wfx200_queue_event), K_NO_WAIT);
+
+	if (event == NULL) {
+		/* TODO */
+		return;
+	}
+
+	if (start_ap->body.status == 0) {
+		LOG_INF("AP started");
+		wfx200_0->sl_context.state |= SL_WFX_AP_INTERFACE_UP;
+		event->ev = WFX200_AP_START_EVENT;
+	} else {
+		LOG_ERR("AP start failed(%d)", start_ap->body.status);
+		event->ev = WFX200_AP_START_FAILED_EVENT;
+	}
+	k_queue_append(&wfx200_0->event_queue, event);
+}
+
+void sl_wfx_stop_ap_callback(sl_wfx_stop_ap_ind_t *stop_ap)
+{
+	struct wfx200_queue_event *event = k_heap_alloc(&wfx200_0->heap,
+							sizeof(struct wfx200_queue_event), K_NO_WAIT);
+
+	ARG_UNUSED(stop_ap);
+
+	if (event == NULL) {
+		/* TODO */
+		return;
+	}
+
+	LOG_DBG("AP Stopped");
+	wfx200_0->sl_context.state &= ~SL_WFX_AP_INTERFACE_UP;
+	event->ev = WFX200_AP_STOP_EVENT;
+	k_queue_append(&wfx200_0->event_queue, event);
+}
+
+void sl_wfx_host_received_frame_callback(sl_wfx_received_ind_t *rx_buffer)
+{
+	struct net_pkt *pkt;
+	int res;
+
+	if (wfx200_0->iface == NULL) {
+		LOG_ERR("Network interface unavailable");
+		return;
+	}
+	switch (wfx200_0->state) {
+	case WFX200_STATE_STA_MODE:
+		if ((rx_buffer->header.info & SL_WFX_MSG_INFO_INTERFACE_MASK) ==
+		    (SL_WFX_SOFTAP_INTERFACE << SL_WFX_MSG_INFO_INTERFACE_OFFSET)) {
+			LOG_WRN("Got ethernet packet from softap interface in sta mode. Dropping packet...");
+			return;
+		}
+		break;
+	case WFX200_STATE_AP_MODE:
+		if ((rx_buffer->header.info & SL_WFX_MSG_INFO_INTERFACE_MASK) ==
+		    (SL_WFX_STA_INTERFACE << SL_WFX_MSG_INFO_INTERFACE_OFFSET)) {
+			LOG_WRN("Got ethernet packet from sta interface in ap mode. Dropping packet...");
+			return;
+		}
+		break;
+	default:
+		LOG_ERR("Network interface not connected. Current state: %s",
+			wfx200_state_to_str(wfx200_0->state));
+	}
+
+	pkt = net_pkt_rx_alloc_with_buffer(wfx200_0->iface, rx_buffer->body.frame_length,
+					   AF_UNSPEC, 0, K_NO_WAIT);
```

Comment:
**TL;DR**
It would be really nice to make timeout here configurable with KConfig, e.g. `K_MSEC(CONFIG_WFX200_RX_ALLOC_TIMEOUT)`

**Some background:**
At least with Zephyr 2.7 with the driver which is quite similar to this one (I'm just using [k_poll_signal_raise  ](https://docs.zephyrproject.org/2.7.0/reference/kernel/other/polling.html?highlight=polling%20api#using-k-poll-signal-raise) for posting events and posting interrupt ready and dedicated rx thread instead of semaphores and workqueue) I observed the following without timeout here:

`net_pkt: Data buffer (576) allocation failed.`
(or other sizes).

And sometimes the OS doesn't recover, continuously showing this message.
With timeout, however, stress test is running for a while already.